README.md

```diff
@@ -1 +1,541 @@
+  - Use `after()` to add an element after a specific element.
+
+- **Cloning nodes:**
+  - Use `.cloneNode(true)` to clone a node, including its child elements.
+
+- **Getting computed styles:**
+  - `getComputedStyle(element)` returns the styles applied to an element from the CSS stylesheet.
+
+- **Data attributes:**
+  - You can access custom data attributes using `dataset`, e.g., `element.dataset.lablablaaa`.
+
+### **Event Handling**
+- **Event targets:**
+  - `e.target`: The actual element that triggered the event.
+  - `e.currentTarget`: The element to which the event listener is attached.
+  
+- **Stopping event propagation:**
+  - `e.stopPropagation()` prevents the event from propagating (bubbling) up to parent elements.
+
+---
+ These concepts form the foundation of understanding how to structure, organize, and manipulate your JavaScript code
+ using OOP principles and how to interact effectively with the DOM using advanced techniques.
+then lets explain some events bro :)
+### 1. **Event Delegation & addEventListener**
+- **Event Delegation**: A technique in JavaScript where you use a single event listener to handle events for multiple child elements. 
+    Instead of adding an event listener to every child,you add it to a parent element.The event listener catches events as they "bubble up"
+    from the child to the parent.
+- **addEventListener Third Parameter**: 
+  - **true**: Enables capturing phase, where the event is caught by the parent before the child.
+  - **false**: Default, uses the bubbling phase, where the event is first caught by the child, then bubbles up to the parent.
+
+### 2. **DOM Traversing**
+- **Selecting Children**:
+  - `h1.children`: Returns an array-like collection of the child elements of the `h1` element.
+  - `h1.firstElementChild` and `h1.lastElementChild`: Return the first and last child elements of `h1`.
+- **Selecting Parents**:
+  - `h1.parentElement`: Returns the parent element of `h1`.
+  - `h1.closest(className)`: Finds the closest ancestor (including the element itself) that matches the selector.
+- **Selecting Siblings**:
+  - `h1.previousElementSibling` and `h1.nextElementSibling`: Return the previous and next sibling elements of `h1`.
+
+### 3. **localStorage**
+- **Set Item**:
+  - `localStorage.setItem('key', string)`: Stores a string in the browser’s local storage. The data persists
+     even after the page is reloaded or the browser is closed.
+  - For objects, convert them to strings using `JSON.stringify(object)`.
+- **Get Item**:
+  - `localStorage.getItem('key')`: Retrieves the string associated with the given key.
+  - If you stored an object, use `JSON.parse(string)` to convert it back to an object.
+- **Remove Item**:
+  - `localStorage.removeItem('key')`: Removes the item with the given key from local storage.
+  - Alternatively, you can set the key to an empty string to effectively clear it.
+
+### 4. **Async JavaScript & Promises**
```

Comment:
### here also please add fetch and then methods to fetch the API from 3d party library